src/service/FileTransferService.ts

```diff
@@ -42,10 +47,62 @@ export class FileTransferService {
                 }
             }).catch((err) => {
                 if (err.code === UNSUPPORTED_MEDIA_TYPE) {
-                    throw new Error('Unsupported file type')
+                    throw new Error('Unsupported file type');
                 } else {
-                    throw new Error(err.message)
+                    throw new Error(err.message);
                 }
             });
     }
+
+    async fileMetada(fileId: string): Promise<FileMetada> {
```

Comment:
What happens if the fileId is null?

Reply:
You can't pass a null to a string arg

Reply:
I am bit on the cross road here. On one hand TS won't allow passing null or undefined if strict type checking is enabled. On the other hand source is transpired to JS which does not care about types and for that reason I tend to have null checks to fail fast

Reply:
Maybe it would be right to have following guidance:
- if code is internal (will never be exposed outside) we relay on TS type checking
- if code is candidate for extraction to NPM like email service then we add extra safety net for JS only projects?

Reply:
I am bit worried that it is so simple to disable string type checks :(

Reply:
Just like in Java when you compile to src code, you can use a bytecode editor to do silly things like pass null ref to string args. It depends on how defensively we want to code. I think that guidance is fair.